client/landing/jetpack-cloud/sections/backups/components/activity-log/index.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { FunctionComponent, useCallback, useEffect } from 'react';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { getHttpData } from 'state/data-layer/http-data';
+import { requestActivityLogs, requestActivityLogsId } from 'state/data-getters';
+import { updateFilter } from 'state/activity-log/actions';
+import { useLocalizedMoment } from 'components/localized-moment';
+import ActivityCard from 'landing/jetpack-cloud/components/activity-card';
+import Filterbar from 'my-sites/activity/filterbar';
+import getActivityLogFilter from 'state/selectors/get-activity-log-filter';
+import Pagination from 'components/pagination';
+import Spinner from 'components/spinner';
+
+/**
+ * Style dependencies
+ */
+import './style.scss';
+
+interface Props {
+	siteId: number;
+	allowRestore: boolean;
+	showDateRange: boolean;
+	pageSize?: number;
+}
+
+//todo: move these typings somewhere else
+interface Activity {
+	activityId: string;
+}
+
+interface Filter {
+	page: number;
+}
+
+const BackupActivityLog: FunctionComponent< Props > = ( {
+	siteId,
+	allowRestore,
+	pageSize = 10,
+} ) => {
+	const dispatch = useDispatch();
+	const moment = useLocalizedMoment();
+	const translate = useTranslate();
+
+	const filter = useSelector< object, Filter >( state => getActivityLogFilter( state, siteId ) );
+	const activities = useSelector< object, Array< Activity > | undefined >(
+		() => getHttpData( requestActivityLogsId( siteId, filter ) ).data
+	);
+
+	const setPage = useCallback(
+		( newPage: number ) => dispatch( updateFilter( siteId, { page: newPage } ) ),
+		[ dispatch, siteId ]
+	);
+
+	useEffect( () => {
+		requestActivityLogs( siteId, filter );
+	}, [ filter, siteId ] );
+
+	const renderPagination = ( key: string, total: number, page: number ) => (
+		<Pagination
+			key={ key }
+			nextLabel={ translate( 'Older' ) }
+			page={ page }
+			pageClick={ setPage }
+			perPage={ pageSize }
+			prevLabel={ translate( 'Newer' ) }
+			total={ total }
+		/>
+	);
+
+	const renderData = ( loadedActivities: Array< Activity > ) => {
+		const actualPage = Math.max(
+			1,
+			Math.min( filter.page, Math.ceil( loadedActivities.length / pageSize ) )
+		);
+		const renderedActivities = loadedActivities
+			.slice( ( actualPage - 1 ) * pageSize, actualPage * pageSize )
+			.map( activity => (
+				<ActivityCard
+					{ ...{
+						key: activity.activityId,
+						moment,
+						activity,
+						allowRestore,
+					} }
+				/>
+			) );
+
+		return (
+			<>
+				{ renderPagination( 'activity-log__pagination-top', loadedActivities.length, actualPage ) }
+				{ renderedActivities }
+				{ renderPagination(
+					'activity-log__pagination-bottom',
+					loadedActivities.length,
+					actualPage
+				) }
+			</>
+		);
+	};
+
+	return (
+		<div className="activity-log">
+			<Filterbar
+				{ ...{
+					siteId,
+					filter,
+					isVisible: true,
+					isLoading: ! activities,
+				} }
+			/>
```

Comment:
Could we move the Filter bar to the upper component `BackupActivityPage`? This way we'll be able to use this component in other parts, on the `BackupPage`, on the `BackupDetailPage` and also on the `BackupActivityLogPage`

**Backup Page**
![image](https://user-images.githubusercontent.com/9832440/78248689-0943ea00-74e5-11ea-8d83-915a039d559e.png)

**Backup Detail Page**
![image](https://user-images.githubusercontent.com/9832440/78248780-28db1280-74e5-11ea-94ae-26a4c19249cc.png)

**Backup Activity Log Page**
![image](https://user-images.githubusercontent.com/9832440/78248962-78214300-74e5-11ea-8ec3-2ac0d17676f2.png)

Reply:
My thinking was possibly allowing it to render some or all of the filter bar depending on the context controlled via props. I.e. `showActivityTypePicker` & `showDatePicker`, but let me iterate on that